### PR TITLE
Feat/breadcrumb

### PR DESCRIPTION
**Pull request by jennieramida**

### CODE REVIEW

File: src/lib/components/DefaultBreadcrumb.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+interface BreadcrumbProps {
+  primaryPage: string;
+  primaryPath: string;
+  secondaryPage?: string;
+  secondaryPath?: string;
+  currentPage: string;
+  mb?: string;
+}
+export const DefualtBreadcrumb = ({
```
Comment (songwongtp) [inline src/lib/components/DefaultBreadcrumb.tsx:14]: *Default

File: src/lib/pages/block-details/components/BlockDetailsTop.tsx [hunk 2]
```diff
@@ -38,67 +38,72 @@ export const BlockDetailsTop = ({ blockData }: BlockDetailsTopProps) => {
...
+              <CustomIcon name="history" boxSize={3} color="pebble.600" />
+              <Text variant="body2" color="text.dark">
+                {dateFromNow(blockData.timestamp)}
+              </Text>
+            </Flex>
+            <DotSeparator />
+            <Text variant="body2" color="text.dark">
+              {formatUTC(blockData.timestamp)}
+            </Text>
+          </Flex>{" "}
```
Comment (songwongtp) [inline src/lib/pages/block-details/components/BlockDetailsTop.tsx:104]: remove {" "}

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/components/DefaultBreadcrumb.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  secondaryPage = "",
+  secondaryPath = "",
+  currentPage,
+  mb = "0",
+}: BreadcrumbProps) => (
+  <Breadcrumb
+    w="full"
+    spacing="4px"
+    mb={mb}
+    separator={<CustomIcon name="chevron-right" boxSize="3" />}
```
Comment (songwongtp) [inline src/lib/components/DefaultBreadcrumb.tsx:26]: boxSize={3}

File: src/lib/components/DefaultBreadcrumb.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import { AppLink } from "./AppLink";
+import { CustomIcon } from "./icon";
+
+interface BreadcrumbProps {
+  primaryPage: string;
+  primaryPath: string;
+  secondaryPage?: string;
+  secondaryPath?: string;
+  currentPage: string;
+  mb?: string;
```
Comment (songwongtp) [inline src/lib/components/DefaultBreadcrumb.tsx:12]: change to number

File: src/lib/pages/contract-details/components/ContractTop.tsx [hunk 5]
```diff
@@ -94,94 +95,115 @@ export const ContractTop = ({ contractData }: ContractTopProps) => {
...
-          </Text>
-          <CopyLink
-            value={contractAddress}
-            amptrackSection="contract_top"
-            type="contract_address"
+    <Flex justify="space-between" mb={6}>
+      <Flex direction="column" w="full">
+        {publicProject.publicDetail && (
+          <DefaultBreadcrumb
+            mb="24px"
```
Comment (songwongtp) [inline src/lib/pages/contract-details/components/ContractTop.tsx:102]: mb={6}

File: src/lib/pages/account-details/index.tsx [hunk 6]
```diff
@@ -86,30 +91,43 @@ const AccountDetailsBody = ({ accountAddress }: AccountDetailsBodyProps) => {
...
-            Wallet Address:
-          </Text>
-          <CopyLink
-            value={accountAddress}
-            amptrackSection="account_top"
-            type="user_address"
+      <Flex direction="column" mb={6}>
+        {publicDetail && (
+          <DefaultBreadcrumb
+            mb="24px"
```
Comment (songwongtp) [inline src/lib/pages/account-details/index.tsx:97]: mb={6}

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/pages/account-details/index.tsx [hunk 7]
```diff
@@ -86,30 +91,46 @@ const AccountDetailsBody = ({ accountAddress }: AccountDetailsBodyProps) => {
...
+                href: `/projects/${publicInfo?.slug}`,
+              },
+              { text: truncate(accountAddress) },
+            ]}
+            mb={6}
           />
+        )}
+        <Flex direction="column" gap={2}>
+          <Flex gap={1} minH="36px" align="center">
+            <CustomIcon name="wallet" boxSize="5" color="secondary.main" />
```
Comment (poomthiti) [inline src/lib/pages/account-details/index.tsx:110]: {5}

File: src/lib/pages/block-details/components/BlockDetailsTop.tsx [hunk 8]
```diff
@@ -38,67 +38,72 @@ export const BlockDetailsTop = ({ blockData }: BlockDetailsTopProps) => {
...
-          </Text>
-          <CopyLink
-            value={blockData.hash.toUpperCase()}
-            amptrackSection="block_details_top"
-            type="block_hash"
-          />
+      <Flex direction="column" gap={2} width="full">
+        <Flex justifyContent="space-between" alignItems="center" width="full">
+          <Flex alignItems="center">
+            <CustomIcon name="block" boxSize="5" color="secondary.main" />
```
Comment (poomthiti) [inline src/lib/pages/block-details/components/BlockDetailsTop.tsx:50]: {5}

File: src/lib/pages/block-details/index.tsx [hunk 9]
```diff
@@ -26,7 +26,12 @@ const BlockDetail = () => {
 
   return (
     <PageContainer>
-      <BackButton />
+      <Breadcrumb
+        items={[
+          { text: "Blocks", href: "/blocks" },
+          { text: blockData?.height.toString() ?? "" },
```
Comment (poomthiti) [inline src/lib/pages/block-details/index.tsx:32]: remove ??

File: src/lib/pages/code-details/index.tsx [hunk 10]
```diff
@@ -41,70 +41,95 @@ const CodeDetailsBody = observer(
...
-                  width={7}
-                  height={7}
-                />
-              )}
-              <Heading as="h5" variant="h5">
-                {localCodeInfo?.name ??
-                  publicProject.publicCodeData?.name ??
-                  codeId}
-              </Heading>
+        <Flex direction="column">
```
Comment (poomthiti) [inline src/lib/pages/code-details/index.tsx:44]: remove, not necessary

File: src/lib/pages/code-details/index.tsx [hunk 11]
```diff
@@ -41,70 +41,95 @@ const CodeDetailsBody = observer(
...
+              {
+                text: publicProject.publicCodeData?.name
+                  ? "Public Projects"
+                  : "Codes",
+                href: publicProject.publicCodeData?.name
+                  ? `/projects`
+                  : "/codes",
+              },
+              {
+                text: publicProject.publicDetail?.name ?? null,
```
Comment (poomthiti) [inline src/lib/pages/code-details/index.tsx:56]: remove ??

File: src/lib/pages/code-details/index.tsx [hunk 12]
```diff
@@ -41,70 +41,95 @@ const CodeDetailsBody = observer(
...
+                href: `/projects/${publicProject.publicCodeData?.slug}`,
+              },
+              { text: codeId.toString() ?? "" },
```
Comment (poomthiti) [inline src/lib/pages/code-details/index.tsx:59]: remove ??

File: src/lib/pages/public-project/components/DetailHeader.tsx [hunk 13]
```diff
@@ -23,48 +14,27 @@ interface DetailHeaderProps {
...
-          fontWeight={700}
-          color="text.dark"
-        >
-          {details?.name}
-        </Text>
-      </BreadcrumbItem>
-    </Breadcrumb>
+      items={[
+        { text: "Public Projects", href: "/projects" },
+        { text: details?.name ?? "" },
```
Comment (poomthiti) [inline src/lib/pages/public-project/components/DetailHeader.tsx:19]: remove ??

**Review by poomthiti — CHANGES_REQUESTED** [review]

File: src/lib/pages/code-details/index.tsx [hunk 14]
```diff
@@ -41,9 +41,25 @@ const CodeDetailsBody = observer(
...
       <>
-        <Flex justify="space-between" mt={6}>
-          <Flex direction="column" gap={1}>
+        <Breadcrumb
+          items={[
+            {
+              text: publicProject.publicCodeData?.name
+                ? "Public Projects"
+                : "Codes",
+              href: publicProject.publicCodeData?.name ? `/projects` : "/codes",
```
Comment (poomthiti) [inline src/lib/pages/code-details/index.tsx:50]: "/projects"

**Review by poomthiti — CHANGES_REQUESTED** [review]

**Review by poomthiti — APPROVED** [review]

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**